build.rs

```diff
@@ -157,3 +171,125 @@ fn main() {
         }
     }
 }
+
+#[cfg(feature = "static_dependencies")]
+fn make_zlib(compiler: &cc::Tool, src_dir: &path::PathBuf, out_dir: &path::PathBuf) {
+    use nix::fcntl;
+    use std::os::fd::AsRawFd;
+
+    // lock README such that if two crates are trying to compile
+    // this at the same time (eg libbpf-rs libbpf-cargo)
+    // they wont trample each other
+    let file = std::fs::File::open(&src_dir.join("zlib/README")).unwrap();
+    let fd = file.as_raw_fd();
+    fcntl::flock(fd, fcntl::FlockArg::LockExclusive).unwrap();
+
+    let status = process::Command::new("./configure")
+        .arg("--static")
+        .arg("--prefix")
+        .arg(".")
+        .arg("--libdir")
+        .arg(out_dir)
+        .env("CC", compiler.path())
+        .env("CFLAGS", compiler.cflags_env())
+        .current_dir(&src_dir.join("zlib"))
+        .status()
+        .expect("could not execute make");
+
+    assert!(status.success(), "make failed");
+
+    let status = process::Command::new("make")
+        .arg("install")
+        .current_dir(&src_dir.join("zlib"))
+        .status()
+        .expect("could not execute make");
+
+    assert!(status.success(), "make failed");
+
+    let status = process::Command::new("make")
+        .arg("distclean")
+        .current_dir(&src_dir.join("zlib"))
+        .status()
+        .expect("could not execute make");
+
+    assert!(status.success(), "make failed");
+}
+
+#[cfg(feature = "static_dependencies")]
+fn make_elfutils(compiler: &cc::Tool, src_dir: &path::PathBuf, out_dir: &path::PathBuf) {
+    use nix::fcntl;
+    use std::os::fd::AsRawFd;
+
+    // lock README such that if two crates are trying to compile
+    // this at the same time (eg libbpf-rs libbpf-cargo)
+    // they wont trample each other
+    let file = std::fs::File::open(&src_dir.join("elfutils/README")).unwrap();
+    let fd = file.as_raw_fd();
+    fcntl::flock(fd, fcntl::FlockArg::LockExclusive).unwrap();
+
+    // location of libz.a
+    let out_lib = format!("-L{}", out_dir.display());
```

Comment:
Nit: move declaration closer to the location of use for the `configure` command?

Reply:
done